test/test_memoryview.py

```diff
@@ -2,11 +2,110 @@
 # coding: utf-8
 
 
+from array import array
 from msgpack import packb, unpackb
+import sys
 
 
-def test_pack_memoryview():
-    data = bytearray(range(256))
-    view = memoryview(data)
-    unpacked = unpackb(packb(view))
-    assert data == unpacked
+# For Python < 3:
+#  - array type only supports old buffer interface
+#  - array.frombytes is not available, must use deprecated array.fromstring
+if sys.version_info[0] < 3:
+    def __make_view(obj):
+        return memoryview(buffer(obj))
+
+    def __make_array(f, data):
+        a = array(f)
+        a.fromstring(data)
+        return a
+else:
+    __make_view = memoryview
+
+    def __make_array(f, data):
+        a = array(f)
+        a.frombytes(data)
+        return a
+
+
+def test_pack_memoryview_byte_8():
+    data = array('l', range(0xff // 4))
+    view = __make_view(data)
+    reconstructed = __make_array('l', unpackb(packb(view)))
+    assert data == reconstructed
```

Comment:
I don't like "reconstructed" approach for this test.
This test will success even if `packb(v, use_bin_type=False)` produce bin.